stellar-multitenant/internal/services/status_change_validator_test.go

```diff
@@ -0,0 +1,160 @@
+				_ = data.CreatePaymentFixture(t, ctx, dbConnectionPool, models.Payment, &data.Payment{
+					Amount:         "50",
+					Status:         data.ReadyPaymentStatus,
+					Disbursement:   disbursement,
+					Asset:          *asset,
+					ReceiverWallet: rw,
+				})
+			},
+			deleteFixtures: func() {
+				data.DeleteAllFixtures(t, ctx, dbConnectionPool)
+			},
+			reqStatus:   tenant.DeactivatedTenantStatus,
+			expectedErr: ErrCannotDeactivateTenant,
+		},
+		{
+			name: "tenant is already deactivated",
+			mockTntManagerFn: func(tntManagerMock *tenant.TenantManagerMock) {
+				tntManagerMock.On("GetTenant", ctx, &tenant.QueryParams{
+					Filters: map[tenant.FilterKey]interface{}{
+						tenant.FilterKeyID: tntID,
+					},
+				}).Return(&tenant.Tenant{ID: tntID, Status: tenant.DeactivatedTenantStatus}, nil).Once()
+			},
+			reqStatus:   tenant.DeactivatedTenantStatus,
+			expectedErr: nil,
+		},
+		{
+			name: "tenant is already activated",
+			mockTntManagerFn: func(tntManagerMock *tenant.TenantManagerMock) {
+				tntManagerMock.On("GetTenant", ctx, &tenant.QueryParams{
+					Filters: map[tenant.FilterKey]interface{}{
+						tenant.FilterKeyID: tntID,
+					},
+				}).Return(&tenant.Tenant{ID: tntID, Status: tenant.ActivatedTenantStatus}, nil).Once()
+			},
+			reqStatus:   tenant.ActivatedTenantStatus,
+			expectedErr: nil,
+		},
+		{
+			name: "cannot activate tenant that isn't deactivated",
+			mockTntManagerFn: func(tntManagerMock *tenant.TenantManagerMock) {
+				tntManagerMock.On("GetTenant", ctx, &tenant.QueryParams{
+					Filters: map[tenant.FilterKey]interface{}{
+						tenant.FilterKeyID: tntID,
+					},
+				}).Return(&tenant.Tenant{ID: tntID, Status: tenant.CreatedTenantStatus}, nil).Once()
+			},
+			reqStatus:   tenant.ActivatedTenantStatus,
+			expectedErr: ErrCannotActivateTenant,
+		},
+		{
+			name: "cannot perform update on tenant to requested status",
+			mockTntManagerFn: func(tntManagerMock *tenant.TenantManagerMock) {
+				tntManagerMock.On("GetTenant", ctx, &tenant.QueryParams{
+					Filters: map[tenant.FilterKey]interface{}{
+						tenant.FilterKeyID: tntID,
+					},
+				}).Return(&tenant.Tenant{ID: tntID, Status: tenant.DeactivatedTenantStatus}, nil).Once()
+			},
+			reqStatus:   tenant.CreatedTenantStatus,
+			expectedErr: ErrCannotPerformStatusUpdate,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.mockTntManagerFn(&tenantManagerMock)
+			if tc.createFixtures != nil {
+				tc.createFixtures()
+			}
+
+			validateStatusErr := ValidateStatus(ctx, &tenantManagerMock, models, tntID, tc.reqStatus)
+			if tc.expectedErr != nil {
+				require.Error(t, validateStatusErr)
+				assert.ErrorContains(t, validateStatusErr, tc.expectedErr.Error())
+			} else {
+				require.NoError(t, validateStatusErr)
+			}
+			if tc.deleteFixtures != nil {
+				tc.deleteFixtures()
+			}
+		})
+
+		tenantManagerMock.AssertExpectations(t)
```

Comment:
💡  For better debuggability and improved test isolation, it'd be better to declare and assert the expectations of the mock inside t.Run. Otherwise, one failed subtest can fail all the subsequent ones.

Reply:
got itt!